force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -57,11 +65,16 @@ public with sharing class FlowControlLoopsCollections {
     public static Integer findMax(Integer num1, Integer num2) {
         // Initialize largestNum as null
         Integer largestNum = null;
-
-        return null; // Replace null with the variable you used to store the result
+        if (num1 == null || num2 == null) {
+            return null;
+        }
+        
+        if (num1 >= num2) {
+            return num1;
+        } else {
+            return num2;
+        }
```

Comment:
Good work. Take a look at the method Math.max. You can solve this with a a lot less code. 

https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_map.htm